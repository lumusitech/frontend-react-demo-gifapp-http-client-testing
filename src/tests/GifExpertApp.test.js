import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'

import GifExpertApp from '../GifExpertApp'

describe('Tensting on <GifExpertApp />', () => {
    test('should match with the snapshot along time', () => {
        const wrapper = shallow(<GifExpertApp />); // con valores por defecto

        expect(wrapper).toMatchSnapshot();
    })
    
    test('should shows a category list', () => {
        const categories = ['One Punch','Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})