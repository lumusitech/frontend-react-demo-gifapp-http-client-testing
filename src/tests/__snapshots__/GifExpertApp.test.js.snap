// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tensting on <GifExpertApp /> should match with the snapshot along time 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</div>
`;

exports[`Tensting on <GifExpertApp /> should shows a category list 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon ball"
    key="Dragon ball"
  />
</div>
`;
